LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,352 @@
+#pragma once
+#include <functional>
+
+template <typename Type>
+class TSLList
+{
+private:
+	int Size;
+	struct FNode
+	{
+		Type Element;
+		FNode* Next = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TSLList()
+	{
+		Size = 0;
+	}
+
+	~TSLList()
+	{
+		Clear();
+	}
+
+	//--------------------------------FIterator
+	class FIterator
+	{
```

Comment:
For encapsulation, you can maybe do:
- Public section for members (Iterator could go here)
- Private section for members 
- Public section for methods
- Private section for methods (If you have any)
Though this is merely a suggestion. I'm not entirely sure as to the proper placement of Iterators and iterator methods in a given class.